macros/src/test/scala/MacroCompilerSpec.scala

```diff
@@ -133,55 +138,69 @@ trait HasSRAMGenerator {
-    generateTestPort(prefix, width, depth, write=false, read=true, readEnable=readEnable)
+  def generateReadPort(prefix: String, width: Option[Int], depth: Option[Int], readEnable: Boolean = false): MacroPort = {
+    generateTestPort(prefix, width, depth, write = false, read = true, readEnable = readEnable)
   }
 
   // Generate a write port for testing.
-  def generateWritePort(prefix: String, width: Int, depth: Int, maskGran: Option[Int] = None, writeEnable: Boolean = true): MacroPort = {
-    generateTestPort(prefix, width, depth, maskGran=maskGran, write=true, read=false, writeEnable=writeEnable)
+  def generateWritePort(prefix: String, width: Option[Int], depth: Option[Int], maskGran: Option[Int] = None, writeEnable: Boolean = true): MacroPort = {
+    generateTestPort(prefix, width, depth, maskGran = maskGran, write = true, read = false, writeEnable = writeEnable)
   }
 
   // Generate a simple read-write port for testing.
-  def generateReadWritePort(prefix: String, width: Int, depth: Int, maskGran: Option[Int] = None): MacroPort = {
+  def generateReadWritePort(prefix: String, width: Option[Int], depth: Option[Int], maskGran: Option[Int] = None): MacroPort = {
     generateTestPort(
-      prefix, width, depth, maskGran=maskGran,
-      write=true, writeEnable=true,
-      read=true, readEnable=false
+      prefix, width, depth, maskGran = maskGran,
+      write = true, writeEnable = true,
+      read = true, readEnable = false
     )
   }
 
   // Generate a "simple" SRAM (active high/positive edge, 1 read-write port).
   def generateSRAM(name: String, prefix: String, width: Int, depth: Int, maskGran: Option[Int] = None, extraPorts: Seq[MacroExtraPort] = List()): SRAMMacro = {
     SRAMMacro(
-      name=name,
-      width=width,
-      depth=depth,
-      family="1rw",
-      ports=Seq(generateReadWritePort(prefix, width, depth, maskGran)),
-      extraPorts=extraPorts
+      name = name,
+      width = width,
+      depth = depth,
+      family = "1rw",
+      ports = Seq(generateReadWritePort(prefix, width, depth, maskGran)),
+      extraPorts = extraPorts
     )
   }
+
+  // Generate a "simple" SRAM group (active high/positive edge, 1 read-write port).
+  def generateSimpleSRAMGroup(prefix: String, mux: Int, depth: Range, width: Range, maskGran: Option[Int] = None, extraPorts: Seq[MacroExtraPort] = List()): SRAMGroup = {
+    SRAMGroup(Seq("mygroup_", "width", "x", "depth", "-", "VT"), "1rw", Seq("svt", "lvt", "ulvt"), mux, depth, width, Seq(generateReadWritePort(prefix, None, None, maskGran)))
+  }
+
+  // 'vt': ('svt','lvt','ulvt'), 'mux': 2,  'depth': range(16,513,8),       'width': range(8,289,2),   'ports': 1
+  // 'vt': ('svt','lvt','ulvt'), 'mux': 4,  'depth': range(32,1025,16),     'width': range(4,145),     'ports': 1}
+  def generateSRAMCompiler(name: String, prefix: String): mdf.macrolib.SRAMCompiler = {
+    SRAMCompiler(name, Seq(
+      generateSimpleSRAMGroup(prefix, 2, Range(16, 513, 8), Range(8, 289, 2)),
```

Comment:
1) Are these meant to be hardcoded, or should this be parameterized?

2) Does it make more sense to have the ranges be inclusive (e.g. `[16, 512]`)? I had to think twice for a bit when seeing numbers like 513 and 1025.

Reply:
1. Its just a helper function, so it can be parameterized however we want it to be, I don't care.
2. I took the numbers from stuff that Colin gave, but you're right I'm not sure whether the numbers are inclusive or not. We should check.